packages/amplify_datastore_plugin_interface/lib/src/types/models/model_association.dart

```diff
@@ -14,14 +14,18 @@
  */
 import 'dart:convert';
 
+import 'package:flutter/foundation.dart';
+
+enum ModelAssociationEnum { HasMany, HasOne, BelongsTo }
+
 class ModelAssociation {
-  final String name;
-  final String targetName;
-  final String associatedName;
-  final String associatedType;
+  ModelAssociationEnum associationType;
+  String targetName; // opt (used in belongsTo)
+  String associatedName; // opt (used in hasMany/hasOne)
+  String associatedType; // opt (used in hasMany/hasOne)
 
-  const ModelAssociation({
-    this.name,
+  ModelAssociation({
+    this.associationType,

```

Comment:
can we not have the const constructor or final fields?

Reply:
thank you, you're correct we should have that - I had removed it previously when the code was different.  I went through other files in models folder and changed to const as well